src/Observable.spec.ts

```diff
@@ -100,4 +100,34 @@ describe('Observable', () => {
             done()
         }, 10)
     })
+
+    it('can scan messages without default value', done => {
+        const pushValues = [1, 2, 3]
+        const expectedValues = [[1], [2, 1], [3, 2]]
+        let result = []
+        Observable.from(pushValues).scan(2).subscribe(
+            value => {
+                result.push(value)
+                expect(value).toEqual(expectedValues.shift())
+                if (result.length === 3) {
+                    setTimeout(done)
+                }
+            }
+        )
+    })
+
+    it('can scan messages with default value', done => {
+        const pushValues = [1, 2, 3]
+        const expectedValues = [[1, 0, 0], [2, 1, 0], [3, 2, 1]]
```

Comment:
Kiedy nowa wartość pojawia się w strumieniu to jest następna po poprzedniej - to brzmi idiotycznie, ale tak należy to sobie wyobrazić. To wynik tego, że strumień to sekwencja tylko rozciągnięta w czasie.
Taki "scan" to operacja utrzymywania sekwencji (już nie rozciągniętej w czasie) wszystkich wartości jakie do tej pory się pojawiły i wybieranie ostatnich `x`.

Wartości w odwrotnej kolejności wymagałyby jeszcze jednej operacji odwrócenia sekwencji.

Jak ten argument Cię nie przekonuje to są jeszcze dwa inne:

Jak patrzysz na taką tablicę to masz wartości uporządkowane względem czasu w naturalnym kierunku czytania.

Ostatecznie: tak właśnie było w wersji pierwszej